### PR TITLE
FE-827 - Update Tooltip Component

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Restyles the Tooltip component
- Deprecates the dark prop
- Horizontal offset is ignored for now, will do arrow positioning potentially in the future
- Makes system props available:  border, shadow, background, padding, typography

Screen Shot 2020-03-06 at 4 00 03 PM
Screen Shot 2020-03-06 at 3 59 52 PM

### How To Test or Verify
- Run storybook with `npm run start:storybook`
- Visit Tooltip stories

### PR Checklist
- [x] Add your changes to `unreleased.md` in the root directory. If you plan on publishing immediately after merging (such as a hotfix) or aren't making changes to a published package, you can ignore this step.
- [x] Provide screenshots or [screen recordings](https://getkap.co/) for any visual changes.
- [ ] Get approval from a UX team member (**#uxfe** or **#design-guild** on Slack) for any visual changes.

### CODE REVIEW

File: packages/matchbox/src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -1,124 +1,166 @@
...
-    horizontalOffset: '0px',
-    forcePosition: false,
-    preferredDirection: {
-      bottom: true,
-      left: false,
-      right: true,
-      top: false
-    }
-  }
+const StyledContent = styled(Box)`
```
Comment (jonambas) [inline packages/matchbox/src/components/Tooltip/Tooltip.js]: Need to change this – using `Box` allows *all* system props, should use a `'div'` instead

File: packages/matchbox/src/components/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -1,124 +1,166 @@
...
+      >
+        {props.children}
+      </Box>
     );
   }
+
+  return (
+    <TooltipOverlay
+      eventDebounce={props.eventDebounce}
+      portalId={props.portalId}
```
Comment (bkemper) [inline packages/matchbox/src/components/Tooltip/Tooltip.js]: I thought there is an accessibility need to render the tooltip DOM with/alongside the button?

Screen Shot 2020-03-18 at 11 15 51 AM

File: packages/matchbox/src/components/Tooltip/Tooltip.js [hunk 3]
```diff
@@ -1,124 +1,166 @@
...
+  horizontalOffset: PropTypes.string, // TODO Either deprecate or style Tooltip tips
+  /**
+   * Disables automatic positioning
+   */
+  forcePosition: PropTypes.bool,
+  children: PropTypes.node,
+  /**
+   * These props are provided automatically through the Overlay component
+   */
+  preferredDirection: PropTypes.shape({
```
Comment (bkemper) [inline packages/matchbox/src/components/Tooltip/Tooltip.js]: follow-up: Shouldn't this be an array?

File: packages/matchbox/src/components/Tooltip/TooltipOverlay.js [hunk 4]
```diff
@@ -1,76 +1,75 @@
...
-  handleMeasurement = () => {
-    this.setState({
-      preferredDirection: getPreferredDirectionFor(this.activator),
-      position: getPositionFor(this.activator)
-    });
-  }
+  return (
+    <>
+      <WindowEvent event="resize" handler={debounce(handleMeasurement, eventDebounce)} />
+      <WindowEvent event="scroll" handler={debounce(handleMeasurement, eventDebounce)} />
```
Comment (bkemper) [inline packages/matchbox/src/components/Tooltip/TooltipOverlay.js]: I know this isn't new, but do we need to call handleMeasurement on resize and scroll events?  The side effect already handles mouseover and mouseout.

File: packages/matchbox/src/components/Tooltip/Tooltip.js (hunk 3 above)
Comment (jonambas) [reply packages/matchbox/src/components/Tooltip/Tooltip.js]: Potentially yeah, also these props don't really need to be defined here and only `top` and `left` are used. Can revise this a bit when we make API updates

File: packages/matchbox/src/components/Tooltip/TooltipOverlay.js (hunk 4 above)
Comment (jonambas) [reply packages/matchbox/src/components/Tooltip/TooltipOverlay.js]: Resize can be removed, but now that we are using fixed positioning, this is the only thing that will hide tooltips if you one is visible before you start scrolling

File: packages/matchbox/src/components/Tooltip/Tooltip.js [hunk 5]
```diff
@@ -139,15 +141,14 @@ Tooltip.propTypes = {
     left: PropTypes.bool,
     right: PropTypes.bool,
   }),
-  /**
-   * Element ID for the portal that will house tooltips. Appends to body if not provided.
-   */
+
   portalId: PropTypes.string,
```
Comment (bkemper) [inline packages/matchbox/src/components/Tooltip/Tooltip.js]: This should probably be removed.

**Pull request merged**